code/modules/atmospherics/machinery/portable/pump.dm

```diff
@@ -155,3 +155,8 @@
 				replace_tank(usr, FALSE)
 				. = TRUE
 	update_icon()
+
+/obj/machinery/portable_atmospherics/pump/CtrlShiftClick(mob/user)
+	if(user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
```

Comment:
You don't want this to work for silicons from further away?
Ignore if intentional